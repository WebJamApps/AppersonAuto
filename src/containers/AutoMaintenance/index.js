import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactResizeDetector from 'react-resize-detector';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';
import PicSlider from '../../components/pic-slider';

export class AutoMaintenance extends Component {
  constructor(props) {
    super(props);
    this.commonUtils = commonUtils;
    this.parentRef = React.createRef();
    this.onResize = this.onResize.bind(this);
    this.state = { width: 320 };
    this.slides = [{ _id: '1', url: '../../static/imgs/photo1.png' },
      { _id: '2', url: '../../static/imgs/photo2.png' },
      { _id: '3', url: '../../static/imgs/photo3.png' },
      { _id: '4', url: '../../static/imgs/photo4.png' },
      { _id: '5', url: '../../static/imgs/photo5.png' },
      { _id: '6', url: '../../static/imgs/photo6.png' },
      { _id: '7', url: '../../static/imgs/photo7.png' },
      { _id: '8', url: '../../static/imgs/photo8.png' },
      { _id: '9', url: '../../static/imgs/photo9.png' },
    ];
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('Auto Maintenance'); }

  onResize(width) { this.setState({ width }); }

  coupon() { // eslint-disable-line class-methods-use-this
    return (
      <div style={{ width: '295px', margin: 'auto' }}>
        <img alt="graphic coupon" src="../static/imgs/banner5.png" />
      </div>
    );
  }

  pageText(marginLeft) { // eslint-disable-line class-methods-use-this
    return (
      <div className="col" style={{ top: '0', paddingRight: '6px', marginLeft }}>
        <h4 style={{ marginTop: '25px', fontWeight: 'bold', fontSize: '16pt' }}>
          Auto Maintenance
        </h4>
        <p>{' '}</p>
        <div>
          <p>
            Are you looking to extend your car’s life? We can help.
            <br />
            Our maintenance service will help keep your car on the road.
          </p>
          <p>
            You’ll receive a
            {' '}
            <b>FREE high mileage / vacation vehicle check over</b>
            . You’ll also receive a
            {' '}
            <b>12-month 12,000-mile warranty</b>
            {' '}
            on all of our services.
          </p>
        </div>
        <div>
          <h4 style={{ marginTop: '20px' }}>Apperson Automotive - Salem, VA Vehicle Care</h4>
          <p><strong>Quality service at a competitive price:</strong></p>
          <ul>
            <li>Tune ups</li>
            <li>Alignments</li>
            <li>Balancing</li>
            <li>Power steering flush</li>
            <li>And much more</li>
          </ul>
          <p><strong>Extend your car’s life with our maintenance service:</strong></p>
          <ul>
            <li>
              Oil changes
            </li>
            <li>
              Tune ups
            </li>
            <li>
              Brake flush
            </li>
            <li>
              Coolant flush
            </li>
          </ul>
          <p><strong>We’ll replace your car’s filters:</strong></p>
          <ul>
            <li>
              Air filters replacement
            </li>
            <li>
              Fuel filters replacement
            </li>
            <li>
              Cabin air filter replacement
            </li>
          </ul>
          <p>
            Same day service may be available.
          </p>
          <p>
            FREE high mileage / vehicle check over service
            <br />
            Our rates are competitive.
            <br />
            Call to set up your maintenance appointment
            <br />
            at Apperson Automotive
            {' '}
            <a style={{ textDecoration: 'none' }} href="tel:5404447337">540-444-7337</a>
          </p>
        </div>
      </div>
    );
  }

  render() {
    const { width } = this.state;
    const marginLeft = width < 900 ? '5px' : '15px';
    return (
      <div className="container-fluid">
        {width < 900
          ? (
            <div className="col" style={{ padding: '1px', paddingRight: '0' }}>
              <div id="familySlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}><PicSlider data={this.slides} /></div>
            </div>
          )
          : null}
        <div className="row">
          {this.pageText(marginLeft)}
          {width >= 900
            ? (
              <div className="col" style={{ padding: '1px', paddingRight: '0' }}>
                <div id="familySlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}><PicSlider data={this.slides} /></div>
                <p style={{ height: '2.5in' }}>{' '}</p>
                {this.coupon()}
              </div>
            )
            : null}
        </div>
        {width < 900 ? this.coupon() : null}
        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} targetDomEl={this.parentRef.current} />

      </div>
    );
  }
}

AutoMaintenance.defaultProps = { homeContent: { title: '', comments: '' } };
AutoMaintenance.propTypes = {
  homeContent: PropTypes.shape({
    title: PropTypes.string,
    comments: PropTypes.string,
  }),
};

export default connect(mapStoreToProps, null)(AutoMaintenance);
