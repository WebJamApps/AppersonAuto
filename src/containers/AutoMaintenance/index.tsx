// @ts-nocheck
import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withResizeDetector } from 'react-resize-detector';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';
import PicSlider from '../../components/pic-slider';
import slidesArr from '../../lib/slides';

export class AutoMaintenance extends Component {
  constructor(props) {
    super(props);
    this.commonUtils = commonUtils;
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('Auto Maintenance', window.screen.width); }

  coupon() { // eslint-disable-line class-methods-use-this
    return (
      <div style={{ width: '295px', margin: 'auto' }}>
        <img id="coupon" alt="graphic coupon" src="../static/imgs/banner5.png" />
      </div>
    );
  }

  filters() {
    const { targetRef, width } = this.props;
    const marginBottom = width < 1162 ? '20px' : '40px';
    return (
      <div ref={targetRef}>
        <p><strong>We’ll replace your car’s filters:</strong></p>
        <ul>
          <li>Air filters replacement</li>
          <li>Fuel filters replacement</li>
          <li>Cabin air filter replacement</li>
        </ul>
        <p>Same day service may be available.</p>
        <p style={{ marginBottom }}>
          FREE high mileage / vehicle check over service
          <br />
          Our rates are competitive.
          <br />
          Call to set up your maintenance appointment
          <br />
          at Apperson Automotive
          {' '}
          <a style={{ textDecoration: 'none' }} href="tel:5404447337">540-444-7337</a>
        </p>
        {width < 1162 ? this.coupon() : null}
      </div>
    );
  }

  quality() { // eslint-disable-line class-methods-use-this
    return (
      <div>
        <h4 style={{ marginTop: '20px' }}>Apperson Automotive - Salem, VA Vehicle Care</h4>
        <p><strong>Quality service at a competitive price:</strong></p>
        <ul>
          <li>Tune ups</li>
          <li>Alignments</li>
          <li>Balancing</li>
          <li>Power steering flush</li>
          <li>And much more</li>
        </ul>
        <p><strong>Extend your car’s life with our maintenance service:</strong></p>
        <ul>
          <li>Oil changes</li>
          <li>Tune ups</li>
          <li>Brake flush</li>
          <li>Coolant flush</li>
        </ul>
        {this.filters()}
      </div>
    );
  }

  pageText(marginLeft) { // eslint-disable-line class-methods-use-this
    return (
      <div className="col" style={{ top: '0', paddingRight: '6px', marginLeft }}>
        <h4 className="pagetitle" style={{ marginTop: '25px', fontWeight: 'bold', fontSize: '16pt' }}>
          Auto Maintenance
        </h4>
        <p>{' '}</p>
        <div>
          <p>
            Are you looking to extend your car’s life? We can help.
            <br />
            Our maintenance service will help keep your car on the road.
          </p>
          <p>
            You’ll receive a
            {' '}
            <b>FREE high mileage / vacation vehicle check over</b>
            . You’ll also receive a
            {' '}
            <b>12-month 12,000-mile warranty</b>
            {' '}
            on all of our services.
          </p>
        </div>
        {this.quality()}
      </div>
    );
  }

  mainPanel(marginLeft) {
    const { width } = this.props;
    return (
      <div className="row">
        {this.pageText(marginLeft)}
        {this.commonUtils.widePics(width, slidesArr, PicSlider, this.coupon, '2.5in')}
      </div>
    );
  }

  render() {
    return this.commonUtils.renderer(slidesArr, this, PicSlider);
  }
}

AutoMaintenance.defaultProps = { homeContent: { title: '', comments: '' }, width: window.screen.width };
AutoMaintenance.propTypes = {
  homeContent: PropTypes.shape({
    title: PropTypes.string,
    comments: PropTypes.string,
  }),
  width: PropTypes.number,
  targetRef: PropTypes.shape({ current: PropTypes.instanceOf(Element) }).isRequired,
};

export default connect(mapStoreToProps, null)(withResizeDetector(AutoMaintenance));
