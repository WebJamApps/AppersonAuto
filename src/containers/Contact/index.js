import React, { Component } from 'react';
import { withResizeDetector } from 
  'react-resize-detector';
import PropTypes from 'prop-types';
import DefaultWideMap from './WideMap';
import DefaultNarrowMap from './NarrowMap';
import commonUtils from '../../lib/commonUtils';

export class Contact extends Component {
  constructor(props) {
    super(props);
    this.commonUtils = commonUtils;
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('Contact', window.screen.width); }

  render() {
    const { targetRef, width } = this.props;
    return (
      <div ref={targetRef} style={{ flexGrow: 1 }}>
        {width >= 1266
          ? (
            <div className="page-content">
              <DefaultWideMap />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
            </div>
          )
          : (
            <div className="page-content">
              <DefaultNarrowMap />
            </div>
          )}
      </div>
    );
  }
}

Contact.propTypes = {
  width: PropTypes.number.isRequired,
  targetRef: PropTypes.shape({ current: PropTypes.element }).isRequired,
};

export default withResizeDetector(Contact);
