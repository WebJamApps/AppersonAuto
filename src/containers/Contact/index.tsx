import { Component, RefObject } from 'react';
import DefaultWideMap from './WideMap';
import DefaultNarrowMap from './NarrowMap';
import commonUtils from 'src/lib/commonUtils';
import { withResizeDetector } from 'react-resize-detector';

interface ContactProps {
  width?: number;
  targetRef: RefObject<HTMLDivElement>;
}

export class Contact extends Component<ContactProps> {
  commonUtils: typeof commonUtils;

  static defaultProps: { width: number; };

  constructor(props: ContactProps) {
    super(props);
    this.commonUtils = commonUtils;
  }

  componentDidMount(): void { this.commonUtils.setTitleAndScroll('Contact', window.screen.width); }

  render(): JSX.Element {
    const { targetRef, width } = this.props;
    return (
      <div ref={targetRef} style={{ flexGrow: 1 }}>
        {width && width >= 1266
          ? (
            <div className="page-content">
              <DefaultWideMap />
              <p style={{ fontSize: '6pt', marginBottom: '0' }}>&nbsp;</p>
            </div>
          )
          : (
            <div className="page-content">
              <DefaultNarrowMap />
            </div>
          )}
      </div>
    );
  }
}

export default withResizeDetector(Contact);
