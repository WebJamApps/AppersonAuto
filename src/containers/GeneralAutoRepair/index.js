import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withResizeDetector } from 'react-resize-detector';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';
import PicSlider from '../../components/pic-slider';
import slidesArr from '../../lib/slides';

export class General extends Component {
  constructor(props) {
    super(props);
    this.commonUtils = commonUtils;
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('General Auto Repair', window.screen.width); }

  coupon() { // eslint-disable-line class-methods-use-this
    return (
      <div style={{ width: '295px', margin: 'auto' }}>
        <img id="coupon" alt="graphic coupon" src="../static/imgs/banner5.png" />
      </div>
    );
  }

  uLists() { // eslint-disable-line class-methods-use-this
    return (
      <div>
        <ul>
          <li>Starter repair / replacement</li>
          <li>Alternator repair / replacement</li>
          <li>Balancing</li>
          <li>Regulators service</li>
          <li>And much more</li>
        </ul>
        <p style={{ marginTop: '30px' }}><strong>Receive 12-month 12,000-mile warranty on our services:</strong></p>
        <ul>
          <li>Brake calipers</li>
          <li>Window motors</li>
          <li>Regulators</li>
          <li>Door locks</li>
        </ul>
      </div>
    );
  }

  love() {
    const { width } = this.props;
    const marginBottom = width < 1162 ? '20px' : '90px';
    return (
      <div>
        <h4 style={{ marginTop: '30px' }}>Apperson Automotive - Salem, VA Vehicle Care</h4>
        <p><strong>You’ll love our general auto repair at competitive prices:</strong></p>
        {this.uLists()}
        <p>We’ll provide you with our quality shuttle service, taking you to and from work.</p>
        <p style={{ marginBottom }}>
          Quality general auto repair services including 12-month 12,000-mile warranty are available.
          {' '}
          To find out more about our services, call Apperson Automotive at
          {' '}
          <a style={{ textDecoration: 'none' }} href="tel:5404447337">540-444-7337</a>
          <br />
          You can also stop by at 1601 Apperson Drive, Salem VA 24153.
        </p>
        {width < 1162 ? this.coupon() : null}
      </div>
    );
  }

  pageText(marginLeft) { // eslint-disable-line class-methods-use-this
    return (
      <div className="col" style={{ top: '0', paddingRight: '6px', marginLeft }}>
        <h4
          className="pagetitle"
          style={{ marginTop: '25px', fontWeight: 'bold', fontSize: '16pt' }}
        >
          General Auto Repair
        </h4>
        <div>
          <p>{' '}</p>
          <p>
            We know your vehicle is important to you. That’s why we’ll do our best to make sure your car is repaired properly the first time!
          </p>
          <p>
            We have more than
            {' '}
            <b>150 years of combined experience</b>
            , so you can count on our service to be trustworthy. Our rates are competitive.
          </p>
          <p>{' '}</p>
          <p>
            <b>FREE high mileage, vacation vehicle check over</b>
            .
          </p>
        </div>
        <p>{' '}</p>
        {this.love()}
      </div>
    );
  }

  mainPanel(marginLeft) {
    const { width } = this.props;
    return (
      <div className="row">
        {this.pageText(marginLeft)}
        {this.commonUtils.widePics(width, slidesArr, PicSlider, this.coupon, '2.5in')}
      </div>
    );
  }

  render() {
    return this.commonUtils.renderer(slidesArr, this, PicSlider);
  }
}

General.defaultProps = { homeContent: { title: '', comments: '' }, width: window.screen.width };
General.propTypes = {
  homeContent: PropTypes.shape({
    title: PropTypes.string,
    comments: PropTypes.string,
  }),
  width: PropTypes.number,
};

export default connect(mapStoreToProps, null)(withResizeDetector(General));
