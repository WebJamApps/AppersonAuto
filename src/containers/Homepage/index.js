import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { withResizeDetector } from 'react-resize-detector';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';
import PicSlider from '../../components/pic-slider';
import slidesArr from '../../lib/slides';

export class Homepage extends Component {
  constructor(props) {
    super(props);
    this.commonUtils = commonUtils;
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('', window.screen.width); }

  getToKnow() { // eslint-disable-line class-methods-use-this
    return (
      <div>
        <p><strong>Get to Know Apperson Automotive:</strong></p>
        <ul>
          <li>
            <a href="/auto-maintenance" title="Apperson Automotive Auto Maintenance">Auto Maintenance</a>
            {' '}
            – Let us handle your oil change, coolant flushes, alignments, and more
          </li>
          <li>
            <a href="/general-auto-repair" title="Apperson Automotive General Auto Repair">General Auto Repair</a>
            {' '}
            – General repairs for includes (not limited to) brake calipers, window motors, regulators, and starters
          </li>
          <li>
            <a href="/major-auto-repair" title="Apperson Automotive Major Auto Repair">Major Auto Repair</a>
            {' '}
            – We’ll handle your steering rack issues, intake gasket repairs, heat and air system problems, and much more
          </li>
        </ul>
        <p>We’ll repair any make or model vehicle you have.</p>
        <p>
          To schedule your FREE high mileage or vacation vehicle check over, call Apperson Automotive at
          {' '}
          <a href="tel:5404447337" style={{ textDecoration: 'none' }}>540-444-7337</a>
          <br />
          You can also stop by at 1601 Apperson Drive, Salem VA 24153.
        </p>
      </div>
    );
  }

  coupon() { // eslint-disable-line class-methods-use-this
    return (
      <div style={{ width: '295px', margin: 'auto' }}>
        <img id="coupon" alt="graphic coupon" src="../static/imgs/banner5.png" />
      </div>
    );
  }

  mechanics() { // eslint-disable-line class-methods-use-this
    return (
      <div>
        <p>
          Are you looking for well qualified mechanics? Look no further!
          <br />
          We have more than
          {' '}
          <strong>150 years of combined experience</strong>
          !
        </p>
        <p>
          Take advantage of our
          <br />
          <strong>FREE high mileage / vacation vehicle check over</strong>
          .
        </p>
        <p>
          <br />
          Receive
          {' '}
          <strong>5% OFF mechanical repairs</strong>
          <br />
          when you present the coupon from this site.
        </p>
        <p>Shuttle service is available to you.</p>
      </div>
    );
  }

  homeText(marginLeft) {
    const { width } = this.props;
    return (
      <div className="col" style={{ top: '0', paddingRight: '6px', marginLeft }}>
        <h4
          className="pagetitle"
          style={{ marginTop: '25px', fontWeight: 'bold', fontSize: '16pt' }}
        >
          Your one stop auto shop where honesty, quality work, and fair pricing are guaranteed
        </h4>
        <p>{' '}</p>
        {this.mechanics()}
        <h4 style={{ marginTop: '40px' }}>Apperson Automotive - Salem, VA Auto Repair Service</h4>
        <p><strong>Quality service for your vehicle (not limited to):</strong></p>
        <ul>
          <li>State inspections</li>
          <li>Alignments</li>
          <li>Tune ups</li>
          <li>Oil changes</li>
          <li>Coolant flush</li>
        </ul>
        {this.getToKnow()}
        {width < 1162 ? this.coupon(targetRef) : null}
      </div>
    );
  }

  mainPanel(marginLeft) {
    const { width } = this.props;
    return (
      <div className="row">
        {this.homeText(marginLeft)}
        {this.commonUtils.widePics(width, slidesArr, PicSlider, this.coupon, '2.5in')}
      </div>
    );
  }

  render() {
    return this.commonUtils.renderer(slidesArr, this, PicSlider);
  }
}

Homepage.defaultProps = { homeContent: { title: '', comments: '' } };
Homepage.propTypes = {
  homeContent: PropTypes.shape({
    title: PropTypes.string,
    comments: PropTypes.string,
  }),
};

export default connect(mapStoreToProps, null)(withResizeDetector(Homepage));
