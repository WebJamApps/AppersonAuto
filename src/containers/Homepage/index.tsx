import { withResizeDetector } from 'react-resize-detector';
import CommonUtils from 'src/lib/commonUtils';
import { PicSlider } from 'src/components/pic-slider';
import slidesArr from 'src/lib/slides';
import { Component, RefObject } from 'react';

interface HomepageProps {
  targetRef?: RefObject<HTMLDivElement>;
  width?: number;
}
export class Homepage extends Component<HomepageProps, unknown> {
  commonUtils = CommonUtils;

  // eslint-disable-next-line @typescript-eslint/no-useless-constructor
  constructor(props:HomepageProps) { super(props); }

  componentDidMount():void { this.commonUtils.setTitleAndScroll('', window.screen.width); }

  getToKnow():JSX.Element { // eslint-disable-line class-methods-use-this
    return (
      <div>
        <p><strong>Get to Know Apperson Automotive:</strong></p>
        <ul>
          <li>
            <a href="/auto-maintenance" title="Apperson Automotive Auto Maintenance">Auto Maintenance</a>
            {' '}
            – Let us handle your oil change, coolant flushes, alignments, and more
          </li>
          <li>
            <a href="/general-auto-repair" title="Apperson Automotive General Auto Repair">General Auto Repair</a>
            {' '}
            – General repairs for includes (not limited to) brake calipers, window motors, regulators, and starters
          </li>
          <li>
            <a href="/major-auto-repair" title="Apperson Automotive Major Auto Repair">Major Auto Repair</a>
            {' '}
            – We’ll handle your steering rack issues, intake gasket repairs, heat and air system problems, and much more
          </li>
        </ul>
        <p>We’ll repair any make or model vehicle you have.</p>
        <p>
          To schedule your FREE high mileage or vacation vehicle check over, call Apperson Automotive at
          {' '}
          <a href="tel:5404447337" style={{ textDecoration: 'none' }}>540-444-7337</a>
          <br />
          You can also stop by at 1601 Apperson Drive, Salem VA 24153.
        </p>
      </div>
    );
  }

  coupon():JSX.Element { // eslint-disable-line class-methods-use-this
    return (
      <div style={{ width: '295px', margin: 'auto' }}>
        <img id="coupon" alt="graphic coupon" src="../static/imgs/banner5.png" />
      </div>
    );
  }

  mechanics():JSX.Element { // eslint-disable-line class-methods-use-this
    return (
      <div>
        <p>
          Are you looking for well qualified mechanics? Look no further!
          <br />
          We have more than
          {' '}
          <strong>150 years of combined experience</strong>
          !
        </p>
        <p>
          Take advantage of our
          <br />
          <strong>FREE high mileage / vacation vehicle check over</strong>
          .
        </p>
        <p>Shuttle service is available to you.</p>
      </div>
    );
  }

  homeText(marginLeft: string):JSX.Element {
    const { targetRef, width } = this.props;
    return (
      <div ref={targetRef} className="col" style={{ top: '0', paddingRight: '6px', marginLeft }}>
        <h4
          className="pagetitle"
          style={{ marginTop: '25px', fontWeight: 'bold', fontSize: '16pt' }}
        >
          Your one stop auto shop where honesty, quality work, and fair pricing are guaranteed
        </h4>
        <p>{' '}</p>
        {this.mechanics()}
        <h4 style={{ marginTop: '40px' }}>Apperson Automotive - Salem, VA Auto Repair Service</h4>
        <p><strong>Quality service for your vehicle (not limited to):</strong></p>
        <ul>
          <li>State inspections</li>
          <li>Alignments</li>
          <li>Tune ups</li>
          <li>Oil changes</li>
          <li>Coolant flush</li>
        </ul>
        {this.getToKnow()}
        {width && width < 1162 ? this.coupon() : null}
      </div>
    );
  }

  mainPanel(marginLeft:string):JSX.Element {
    const { targetRef, width } = this.props;
    return (
      <div ref={targetRef} className="row">
        {this.homeText(marginLeft)}
        {this.commonUtils.widePics(width || 300, slidesArr, PicSlider, this.coupon, '2.5in')}
      </div>
    );
  }

  render():JSX.Element {
    return this.commonUtils.renderer(slidesArr, this, PicSlider);
  }
}

export default withResizeDetector(Homepage);
