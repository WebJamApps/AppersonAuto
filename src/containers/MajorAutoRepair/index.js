import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import ReactResizeDetector from 'react-resize-detector';
import mapStoreToProps from '../../redux/mapStoreToProps';
import commonUtils from '../../lib/commonUtils';
import PicSlider from '../../components/pic-slider';

export class Major extends Component {
  constructor(props) {
    super(props);
    this.commonUtils = commonUtils;
    this.parentRef = React.createRef();
    this.onResize = this.onResize.bind(this);
    this.state = { width: 320 };
    this.slides = [{ _id: '1', url: '../../static/imgs/photo1.png' },
      { _id: '2', url: '../../static/imgs/photo2.png' },
      { _id: '3', url: '../../static/imgs/photo3.png' },
      { _id: '4', url: '../../static/imgs/photo4.png' },
      { _id: '5', url: '../../static/imgs/photo5.png' },
      { _id: '6', url: '../../static/imgs/photo6.png' },
      { _id: '7', url: '../../static/imgs/photo7.png' },
      { _id: '8', url: '../../static/imgs/photo8.png' },
      { _id: '9', url: '../../static/imgs/photo9.png' },
    ];
  }

  componentDidMount() { this.commonUtils.setTitleAndScroll('Major Auto Repair'); }

  onResize(width) { this.setState({ width }); }

  getToKnow() { // eslint-disable-line class-methods-use-this
    return (
      <div>
        <p><strong>Get to Know Apperson Automotive:</strong></p>
        <ul>
          <li>
            <a href="/auto-maintenance" title="Apperson Automotive Auto Maintenance">Auto Maintenance</a>
            {' '}
            – Let us handle your oil change, coolant flushes, alignments, and more
          </li>
          <li>
            <a href="/general-auto-repair" title="Apperson Automotive General Auto Repair">General Auto Repair</a>
            {' '}
            – General repairs for includes (not limited to) brake calipers, window motors, regulators, and starters
          </li>
          <li>
            <a href="/major-auto-repair" title="Apperson Automotive Major Auto Repair">Major Auto Repair</a>
            {' '}
            – We’ll handle your steering rack issues, intake gasket repairs, heat and air system problems, and much more
          </li>
        </ul>
        <p>We’ll repair any make or model vehicle you have.</p>
        <p>
          To schedule your FREE high mileage or vacation vehicle check over, call Apperson Automotive at
          {' '}
          <a href="tel:5404447337" style={{ textDecoration: 'none' }}>540-444-7337</a>
          <br />
          You can also stop by at 1601 Apperson Drive, Salem VA 24153.
        </p>
      </div>
    );
  }

  coupon() { // eslint-disable-line class-methods-use-this
    return (
      <div style={{ width: '295px', margin: 'auto' }}>
        <img alt="graphic coupon" src="../static/imgs/banner5.png" />
      </div>
    );
  }

  pageText(marginLeft) { // eslint-disable-line class-methods-use-this
    return (
      <div className="col" style={{ top: '0', paddingRight: '6px', marginLeft }}>
        <h4 style={{ marginTop: '25px', fontWeight: 'bold', fontSize: '16pt' }}>
          Your one stop auto shop where honesty, quality work, and fair pricing are guaranteed
        </h4>
        <p>{' '}</p>
        <div>
          <p>
            Are you looking for well qualified mechanics? Look no further!
            <br />
            We have more than
            {' '}
            <strong>150 years of combined experience</strong>
            !
          </p>
          <p>
            Take advantage of our
            <br />
            <strong>FREE high mileage / vacation vehicle check over</strong>
            .
          </p>
          <p>
            <br />
            Receive
            {' '}
            <strong>5% OFF mechanical repairs</strong>
            <br />
            when you present the coupon from this site.
          </p>
          <p>Shuttle service is available to you.</p>
        </div>
        <h4 style={{ marginTop: '40px' }}>Apperson Automotive - Salem, VA Auto Repair Service</h4>
        <p><strong>Quality service for your vehicle (not limited to):</strong></p>
        <ul>
          <li>State inspections</li>
          <li>Alignments</li>
          <li>Tune ups</li>
          <li>Oil changes</li>
          <li>Coolant flush</li>
        </ul>
        {this.getToKnow()}
      </div>
    );
  }

  render() {
    const { width } = this.state;
    const marginLeft = width < 900 ? '5px' : '15px';
    return (
      <div className="container-fluid">
        {width < 900
          ? (
            <div className="col" style={{ padding: '1px', paddingRight: '0' }}>
              <div id="familySlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}><PicSlider data={this.slides} /></div>
            </div>
          )
          : null}
        <div className="row">
          {this.pageText(marginLeft)}
          {width >= 900
            ? (
              <div className="col" style={{ padding: '1px', paddingRight: '0' }}>
                <div id="familySlideshowWide" style={{ display: 'flex', flexDirection: 'column' }}><PicSlider data={this.slides} /></div>
                <p style={{ height: '2.5in' }}>{' '}</p>
                {this.coupon()}
              </div>
            )
            : null}
        </div>
        {width < 900 ? this.coupon() : null}
        <ReactResizeDetector handleWidth handleHeight onResize={this.onResize} targetDomEl={this.parentRef.current} />

      </div>
    );
  }
}

Major.defaultProps = { homeContent: { title: '', comments: '' } };
Major.propTypes = {
  homeContent: PropTypes.shape({
    title: PropTypes.string,
    comments: PropTypes.string,
  }),
};

export default connect(mapStoreToProps, null)(Major);
