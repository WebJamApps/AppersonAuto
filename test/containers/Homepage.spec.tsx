import React from 'react';
import { shallow } from 'enzyme';
import { Homepage } from '../../src/containers/Homepage';

describe('Home', () => {
  let wrapper: any;
  beforeEach(() => {
    const width = 1300;
    const height = 1300;
    const targetRef: any = {};
    wrapper = shallow(<Homepage targetRef={targetRef} width={width} height={height} />);
  });
  it('Renders the Homepage', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('renders with cellphone width and has coupon', () => {
    const width = 300;
    const height = 300;
    const targetRef: any = {};
    const wrapper2 = shallow(<Homepage targetRef={targetRef} width={width} height={height} />);
    wrapper2.update();
    const i = wrapper2.find('img#coupon');
    expect(i).toBeDefined();
  });
  it('renders widescreen width and has coupon', () => {
    const i = wrapper.find('img#coupon');
    expect(i).toBeDefined();
  });
});
