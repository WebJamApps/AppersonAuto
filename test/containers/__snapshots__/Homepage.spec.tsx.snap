// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Renders the Homepage 1`] = `ShallowWrapper {}`;
